recipes/hexl/all/conanfile.py

```diff
@@ -0,0 +1,129 @@
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "experimental": [True, False],
+        "fpga_compatibility_dyadic_multiply": [True, False],
+        "fpga_compatibility_keyswitch":  [True, False]
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "experimental": False,
+        "fpga_compatibility_dyadic_multiply": False,
+        "fpga_compatibility_keyswitch": False
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("hexl-{0}".format(self.version), self._source_subfolder)
+
+    def build_requirements(self):
+        self.build_requires("cmake/3.22.0")
+
+    def requirements(self):
+        self.requires("cpu_features/0.7.0")
+
+        if self.settings.build_type == "Debug":
+            self.requires("easyloggingpp/9.97.0")
+
+    def validate(self):
+        tools.check_min_cppstd(self, "17")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self);
+
+
+        self._cmake.definitions["HEXL_BENCHMARK"] = False
+        self._cmake.definitions["HEXL_TESTING"] = False
+        self._cmake.definitions["HEXL_EXPERIMENTAL"] = self.options.experimental
+
+
+        if self.options.fpga_compatibility_dyadic_multiply and self.options.fpga_compatibility_keyswitch:
+            self._cmake.definitions["HEXL_FPGA_COMPATIBILITY"] = 3
+        elif self.options.fpga_compatibility_dyadic_multiply:
+            self._cmake.definitions["HEXL_FPGA_COMPATIBILITY"] = 1
+        elif self.options.fpga_compatibility_keyswitch:
+            self._cmake.definitions["HEXL_FPGA_COMPATIBILITY"] = 2
+        else:
+            self._cmake.definitions["HEXL_FPGA_COMPATIBILITY"] = 0
+
+        self._cmake.definitions["HEXL_SHARED_LIB"] = self.options.shared
+        
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.verbose = True
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.cmake")
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
```

Comment:
```suggestion
        rmdir(self, os.path.join(self.package_folder, "share"))
```
from conan.tools.files import rmdir
https://docs.conan.io/en/latest/reference/conanfile/tools/files/basic.html#conan-tools-files-rmdir

Reply:
Done